### PR TITLE
KAADO-29: Create user service

### PR DESCRIPTION
**Pull request by nasudon**
- Create user service
- Add bcrypt to package

### CODE REVIEW

File: src/services/userService.ts [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+import { stringToObjectId } from '../lib/utils';
+
+import TypeModel from '@/models/Type';
+import UserModel, { IUser } from '@/models/User';
+
+export async function getUsers() {
+    try {
+        await connectDB();
+
+        const users = await UserModel.find<IUser>({}, 'name');
```
Comment (tsoibet) [inline src/services/userService.ts:14]: Do we not need ids of users?

File: src/services/userService.ts [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+                name: user.name,
+                _id: user._id,
+            },
+        };
+    } catch (error) {
+        return { error };
+    }
+}
+
+export async function getUser(userID = '123456789012') {
```
Comment (tsoibet) [inline src/services/userService.ts:45]: What is the use of default value `userID = '123456789012'`?

File: src/services/userService.ts [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    try {
+        await connectDB();
+
+        const user = await UserModel.findOne<IUser>({ name });
+        if (!user) {
+            return { error: 'User not found' };
+        }
+        return { result: await bcrypt.compare(password, user.password) };
+    } catch (error) {
+        return { error: 'Deletion failed' };
```
Comment (tsoibet) [inline src/services/userService.ts]: Error message is not correct.

**Review by tsoibet — CHANGES_REQUESTED** [review]
Please correct the error message in authenticateUser function.

File: src/services/userService.ts (hunk 1 above)
Comment (nasudon) [reply src/services/userService.ts:14]: In mongodb, the id will be returned even you don't specify

File: src/services/userService.ts (hunk 3 above)
Comment (nasudon) [reply src/services/userService.ts]: Thank you for pointing this out. Updated: https://github.com/tsoibet/kaado/pull/8/commits/17fe8e180ecfad82b441055e8f5a3a96c360d415

**Review by tsoibet — APPROVED** [review]
LGTM

**Pull request merged**